### DISCUSSION TITLE
How to batch data for GCNConv?

### DISCUSSION THREAD

**Discussion by CombatOrpheus** [pyg-team/pytorch_geometric#9531] [Q&A]
I'm trying to run some experiments using the GCNConv model, but, whenever I try to execute batched, data, I get the following message: `ValueError: 'GCNConv' received a tuple of node features as input while this layer does not support bipartite message passing. Please try other layers such as 'SAGEConv' or 'GraphConv' instead`.

I have already padded my node features so that they have the same length across the dataset, but the issue remains. Here's the model:
```python
model = torch_geometric.nn.models.GCN(in_channels=data.dataset[0].num_features, hidden_channels=2, out_channels=1, num_layers=3)
```
The usual form of my batches:
```python
DataBatch(x=[32], edge_index=[2, 192], edge_attr=[192, 1], y=[32], batch=[160], ptr=[33])
```
And a single `Data` element:
```python
Data(x=[46, 15], edge_index=[2, 113], edge_attr=[113, 1], y=10.0)
```

I do feel like there's a simple fix for this that I am not able to see, so any help or suggestions is greatly appreciated.

Thanks!

**Comment by rusty1s** [comment]
Can you check why your input `x` into `GCN` is a tuple rather than a simple `torch.Tensor`?

**Comment by CombatOrpheus** [comment]
So, the problem was on my end: I wasn't pre-processing the data properly, which probably explains why I ended up with tuples rather than tensors when creating the `DataLoader`. As is usually the case, the problem was between the chair and the keyboard.
